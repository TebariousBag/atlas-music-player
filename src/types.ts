// api response types from provided api
export interface PlaylistSong {
  id: string;
  title: string;
  artist: string;
  genre: string;
  duration: number;
}

// song extends from playlist giving the cover art and link to song
export interface Song extends PlaylistSong {
  cover: string;
  song: string;
}

// optional, skipping for now
export interface Lyrics {
  lyrics: string;
}

// compnent prop types
export interface CoverArtProps {
  selectedSong?: Song | null;
}

export interface SongTitleProps {
  selectedSong?: Song | null;
}

export interface PlayControlsProps {
  playSpeed: number;
  canGoBack: boolean;
  canGoForward: boolean;
  onChangeSpeed: () => void;
  ongoBack: () => void;
  ongoForward: () => void;
}
