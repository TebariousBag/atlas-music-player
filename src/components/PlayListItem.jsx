export default function PlayListItem() {
  return (
    <div className="flex cursor-pointer items-center justify-between rounded-lg hover:bg-gray-100">
      <div>
        <p className="font-medium">Painted in Blue</p>
        <p className="text-sm text-slate-500">Soul Canvas</p>
      </div>
      <div>
        <p className="text-sm text-slate-500">5:55</p>
      </div>
    </div>
  );
}
