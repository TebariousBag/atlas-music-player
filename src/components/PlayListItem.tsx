import { PlaylistItemProps } from "../types";

export default function PlayListItem({
  title,
  artist,
  duration,
}: PlaylistItemProps) {
  return (
    <div className="hover:bg-picton-blue flex cursor-pointer items-center justify-between rounded-lg p-2">
      <div>
        <p className="text-french-blue font-medium dark:text-gray-300">
          {title}
        </p>
        <p className="text-sm text-slate-500 dark:text-gray-400">{artist}</p>
      </div>
      <div>
        <p className="text-sm text-slate-500">{duration}</p>
      </div>
    </div>
  );
}
