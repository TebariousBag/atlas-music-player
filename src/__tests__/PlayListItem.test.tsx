import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import PlayListItem from '../components/PlayListItem'
import { PlaylistSong } from '../types'

describe('PlayListItem', () => {
  const mockSong: PlaylistSong = {
    id: '1',
    title: 'Test Song',
    artist: 'Test Artist',
    genre: 'Test Genre',
    duration: 180
  }

  const mockOnClick = () => {}

  it('renders unselected song', () => {
    const { container } = render(
      <PlayListItem 
        song={mockSong} 
        isSelected={false} 
        onClick={mockOnClick} 
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders selected song', () => {
    const { container } = render(
      <PlayListItem 
        song={mockSong} 
        isSelected={true} 
        onClick={mockOnClick} 
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders song with long title and artist', () => {
    const longSong: PlaylistSong = {
      id: '2',
      title: 'This is a very long song title that might wrap to multiple lines',
      artist: 'This is a very long artist name that might also wrap',
      genre: 'Test Genre',
      duration: 300
    }

    const { container } = render(
      <PlayListItem 
        song={longSong} 
        isSelected={false} 
        onClick={mockOnClick} 
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders song with short duration', () => {
    const shortSong: PlaylistSong = {
      id: '3',
      title: 'Short Song',
      artist: 'Short Artist',
      genre: 'Test Genre',
      duration: 45
    }

    const { container } = render(
      <PlayListItem 
        song={shortSong} 
        isSelected={false} 
        onClick={mockOnClick} 
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders song with long duration', () => {
    const longSong: PlaylistSong = {
      id: '4',
      title: 'Long Song',
      artist: 'Long Artist',
      genre: 'Test Genre',
      duration: 3661
    }

    const { container } = render(
      <PlayListItem 
        song={longSong} 
        isSelected={true} 
        onClick={mockOnClick} 
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
