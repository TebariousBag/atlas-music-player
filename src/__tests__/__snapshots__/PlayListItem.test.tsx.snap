// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders selected song 1`] = `
<div
  class="flex cursor-pointer items-center justify-between rounded-lg p-2 transition-colors bg-picton-blue dark:bg-gray-700"
>
  <div>
    <p
      class="text-french-blue font-medium dark:text-gray-300"
    >
      Test Song
    </p>
    <p
      class="text-sm text-slate-500 dark:text-gray-400"
    >
      Test Artist
    </p>
  </div>
  <div>
    <p
      class="text-sm text-slate-500"
    >
      3:00
    </p>
  </div>
</div>
`;

exports[`PlayListItem > renders song with long duration 1`] = `
<div
  class="flex cursor-pointer items-center justify-between rounded-lg p-2 transition-colors bg-picton-blue dark:bg-gray-700"
>
  <div>
    <p
      class="text-french-blue font-medium dark:text-gray-300"
    >
      Long Song
    </p>
    <p
      class="text-sm text-slate-500 dark:text-gray-400"
    >
      Long Artist
    </p>
  </div>
  <div>
    <p
      class="text-sm text-slate-500"
    >
      61:01
    </p>
  </div>
</div>
`;

exports[`PlayListItem > renders song with long title and artist 1`] = `
<div
  class="flex cursor-pointer items-center justify-between rounded-lg p-2 transition-colors hover:bg-picton-blue dark:hover:bg-gray-900"
>
  <div>
    <p
      class="text-french-blue font-medium dark:text-gray-300"
    >
      This is a very long song title that might wrap to multiple lines
    </p>
    <p
      class="text-sm text-slate-500 dark:text-gray-400"
    >
      This is a very long artist name that might also wrap
    </p>
  </div>
  <div>
    <p
      class="text-sm text-slate-500"
    >
      5:00
    </p>
  </div>
</div>
`;

exports[`PlayListItem > renders song with short duration 1`] = `
<div
  class="flex cursor-pointer items-center justify-between rounded-lg p-2 transition-colors hover:bg-picton-blue dark:hover:bg-gray-900"
>
  <div>
    <p
      class="text-french-blue font-medium dark:text-gray-300"
    >
      Short Song
    </p>
    <p
      class="text-sm text-slate-500 dark:text-gray-400"
    >
      Short Artist
    </p>
  </div>
  <div>
    <p
      class="text-sm text-slate-500"
    >
      0:45
    </p>
  </div>
</div>
`;

exports[`PlayListItem > renders unselected song 1`] = `
<div
  class="flex cursor-pointer items-center justify-between rounded-lg p-2 transition-colors hover:bg-picton-blue dark:hover:bg-gray-900"
>
  <div>
    <p
      class="text-french-blue font-medium dark:text-gray-300"
    >
      Test Song
    </p>
    <p
      class="text-sm text-slate-500 dark:text-gray-400"
    >
      Test Artist
    </p>
  </div>
  <div>
    <p
      class="text-sm text-slate-500"
    >
      3:00
    </p>
  </div>
</div>
`;
